eZ/Publish/API/Repository/Tests/SettingServiceTest.php

```diff
@@ -0,0 +1,217 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\API\Repository\Tests;
+
+use eZ\Publish\API\Repository\Exceptions\InvalidArgumentException;
+use eZ\Publish\API\Repository\Exceptions\NotFoundException;
+use eZ\Publish\API\Repository\SettingService;
+use eZ\Publish\API\Repository\Values\Setting\Setting;
+use eZ\Publish\API\Repository\Values\Setting\SettingCreateStruct;
+use eZ\Publish\API\Repository\Values\Setting\SettingUpdateStruct;
+
+/**
+ * Test case for operations in the SettingService using in memory storage.
+ *
+ * @see \eZ\Publish\API\Repository\SettingService
+ * @group integration
+ * @group setting
+ */
+class SettingServiceTest extends BaseTest
+{
+    /** @var \eZ\Publish\API\Repository\PermissionResolver */
+    protected $permissionResolver;
+
+    /** @var \eZ\Publish\API\Repository\SettingService */
+    protected $settingService;
+
+    protected function getSettingService(): SettingService
+    {
+        $container = $this->getSetupFactory()->getServiceContainer();
+        /** @var \eZ\Publish\API\Repository\SettingService $settingService */
+        $settingService = $container->get('ezpublish.api.service.setting');
+
+        return $settingService;
+    }
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $repository = $this->getRepository(false);
+        $this->permissionResolver = $repository->getPermissionResolver();
+    }
+
+    /**
+     * Test for the newSettingCreateStruct() method.
+     *
+     * @see \eZ\Publish\API\Repository\SettingService::newSettingCreateStruct()
+     */
+    public function testNewSettingCreateStruct()
+    {
+        $settingService = $this->getSettingService();
+        $settingCreate = $settingService->newSettingCreateStruct();
+
+        $this->assertInstanceOf(SettingCreateStruct::class, $settingCreate);
```

Comment:
This does not provide any value as it tests if the language works. It will never fail because it would crash with fatal error first.
Maybe check for default values of the struct or set some values and assert that?